clean_code_block/src/main/java/Airport.java

```diff
@@ -0,0 +1,124 @@
+        PassengerPlane planeWithMaxCapacity = passengerPlanes.get(0);
+        for (int i = 0; i < passengerPlanes.size(); i++) {
+            if (passengerPlanes.get(i).getPassengersCapacity() > planeWithMaxCapacity.getPassengersCapacity()) {
+                planeWithMaxCapacity = passengerPlanes.get(i);
+            }
+        }
+        return planeWithMaxCapacity;
+    }
+
+    public List<MilitaryPlane> getTransportMilitaryPlanes() {
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
+        List<MilitaryPlane> militaryPlanes = getMilitaryPlanes();
+        for (MilitaryPlane plane : militaryPlanes) {
+            if (plane.getType() == MilitaryType.TRANSPORT) {
+                transportMilitaryPlanes.add(plane);
+            }
+        }
+        return transportMilitaryPlanes;
+    }
+
+    public List<MilitaryPlane> getBomberMilitaryPlanes() {
+        List<MilitaryPlane> bomberMilitaryPlanes = new ArrayList<>();
+        List<MilitaryPlane> militaryPlanes = getMilitaryPlanes();
+        for (MilitaryPlane plane : militaryPlanes) {
+            if (plane.getType() == MilitaryType.BOMBER) {
+                bomberMilitaryPlanes.add(plane);
+            }
+        }
+        return bomberMilitaryPlanes;
+    }
+
+    public List<ExperimentalPlane> getExperimentalPlanes() {
+        List<ExperimentalPlane> experimentalPlanes = new ArrayList<>();
+        for (Plane plane : planes) {
+            if (plane instanceof ExperimentalPlane) {
+                experimentalPlanes.add((ExperimentalPlane) plane);
+            }
+        }
+        return experimentalPlanes;
+    }
+
+    public Airport sortByMaxDistance() {
+        Collections.sort(planes, new Comparator<Plane>() {
+            public int compare(Plane o1, Plane o2) {
+                return o1.getMaxFlightDistance() - o2.getMaxFlightDistance();
+            }
+        });
+        return this;
+    }
+
+    public Airport sortByMaxSpeed() {
+        Collections.sort(planes, new Comparator<Plane>() {
+            public int compare(Plane o1, Plane o2) {
+                return o1.getMaxSpeed() - o2.getMaxSpeed();
+            }
+        });
+        return this;
+    }
+
+    public Airport sortByMaxLoadCapacity() {
+        Collections.sort(planes, new Comparator<Plane>() {
+            public int compare(Plane o1, Plane o2) {
+                return o1.getMaxLoadCapacity() - o2.getMaxLoadCapacity();
+            }
+        });
+        return this;
+    }
+
+    public List<? extends Plane> getPlanes() {
+        return planes;
+    }
+
+    private void print(Collection<? extends Plane> collection) {
+        for (Plane plane : collection) {
+            System.out.println(plane);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "Airport{" +
+                "Planes=" + planes.toString() +
+                '}';
+    }
+
+    public Airport(List<? extends Plane> planes) {
```

Comment:
please move constructor to be a first method in the class

Reply:
removed